openapi_python_client/parser/properties/property.py

```diff
@@ -32,21 +32,34 @@ class Property:
     def __attrs_post_init__(self) -> None:
         object.__setattr__(self, "python_name", utils.to_valid_python_identifier(utils.snake_case(self.name)))
 
-    def get_type_string(self, no_optional: bool = False) -> str:
+    def get_base_type_string(self) -> str:
+        return self._type_string
+
+    def get_type_string(self, no_optional: bool = False, query_parameter: bool = False) -> str:
         """
         Get a string representation of type that should be used when declaring this property
 
         Args:
             no_optional: Do not include Optional or Unset even if the value is optional (needed for isinstance checks)
+            query_parameter: True if the property's type is being used for a query parameter
         """
-        type_string = self._type_string
+        type_string = self.get_base_type_string()
         if no_optional:
-            return self._type_string
-        if self.nullable:
-            type_string = f"Optional[{type_string}]"
-        if not self.required:
-            type_string = f"Union[Unset, {type_string}]"
-        return type_string
+            return type_string
+        if self.required:
+            if self.nullable:
+                return f"Optional[{type_string}]"
+            else:
+                return type_string
+        else:
+            if self.nullable:
+                return f"Union[Unset, None, {type_string}]"
+            else:
+                if query_parameter:
+                    # For query parameters, None has the same meaning as Unset
+                    return f"Union[Unset, None, {type_string}]"
+                else:
+                    return f"Union[Unset, {type_string}]"
```

Comment:
I can't include it in a suggestion because of the way GitHub split the lines, but you could make this first case the same if check as `no_optional`. This is basically the same change I suggested for `UnionProperty`.

```suggestion
        if self.required and not self.nullable:
            return type_string
        if query_parameter or (self.nullable and not self.required):
            # For query parameters, nullable has the same meaning as required
            return f"Union[Unset, None, {type_string}]"
        if self.nullable:
            return f"Optional[{type_string}]"
        return f"Union[Unset, {type_string}]"
```